core/tools/objectinspector/objectinspector.cpp

```diff
@@ -206,3 +210,61 @@ void ObjectInspector::scanForConnectionIssues()
         }
     }
 }
+
+void ObjectInspector::scanForThreadAffinityIssues()
+{
+    const auto &objects = Probe::instance()->allQObjects();
+
+    QMutexLocker lock(Probe::objectLock());
+    for (const auto object : objects) {
+        if (!Probe::instance()->isValidObject(object)) {
+            continue;
+        }
+
+        const auto objectName = Util::displayString(object);
+        if (object == object->thread()) {
+            const auto objectName = Util::displayString(object);
+            Problem problem;
+            problem.severity = Problem::Warning;
+            problem.description = QStringLiteral("The thread %1 has affinity with itself.").arg(objectName);
+            problem.object = ObjectId(object);
+            // p.location = bindingNode->sourceLocation(); //TODO can we get source locations of new-statements?
+            problem.problemId = QStringLiteral("com.kdab.GammaRay.ObjectInspector.ThreadAffinityCheck.Self.%1")
+                    .arg(QString::number(reinterpret_cast<quintptr>(object)));
+            problem.findingCategory = Problem::Scan;
+            ProblemCollector::addProblem(problem);
+        }
+
+        const auto parent = object->parent();
+        if (parent == nullptr) {
+            continue;
+        }
+
+        const auto parentName = Util::displayString(parent);
+        if (object->thread() != parent->thread()) {
+            Problem problem;
+            problem.severity = Problem::Warning;
+            problem.description = QStringLiteral("The object %1 doesn't have the same thread affinity as its parent %2.").arg(objectName, parentName);
+            problem.object = ObjectId(object);
+            // p.location = bindingNode->sourceLocation(); //TODO can we get source locations of new-statements?
+            problem.problemId = QStringLiteral("com.kdab.GammaRay.ObjectInspector.ThreadAffinityCheck.%1:%2")
+                    .arg(QString::number(reinterpret_cast<quintptr>(object)),
+                         QString::number(reinterpret_cast<quintptr>(parent)));
+            problem.findingCategory = Problem::Scan;
+            ProblemCollector::addProblem(problem);
+        }
+
+        if (parent->inherits("QThread") && object->thread() != object->parent()) {
```

Comment:
qobject_cast<QThread*>(parent) would be an alternative to inherits that doesn't require string comparison